firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1770,14 +1770,28 @@ class BrowserViewController: UIViewController,
         }
 
         if webViewStatus == .finishedNavigation {
-            if tab !== tabManager.selectedTab, let webView = tab.webView {
+            if tab !== tabManager.selectedTab,
+                let webView = tab.webView,
+                tab.screenshot == nil {
```

Comment:
note: Wouldn't adding nil here mean users will still see solid screenshots?

Reply:
It will only enter this if there is no screenshot. Meaning it will always trigger when launching a new tab in the background but will not trigger for any weird scenarios that cause the webview to reload without rendering properly, something that was likely triggering the valid screenshots to be overwritten by black screenshots in the past. Can't verify for sure tho as it's hard to reproduce.